reliability-engineering/terraform/modules/concourse-team-iam-credentials/files/lambda_rotate_sts.py

```diff
@@ -0,0 +1,48 @@
+import boto3
+import json
+import os
+
+print('Loading function')
+
+ssm = boto3.client('ssm')
+sts = boto3.client('sts')
+
+
+def lambda_handler(event, context):
+    print("Received event: " + json.dumps(event, indent=2))
+
+    deployment = os.environ.get('DEPLOYMENT')
+    key_id = os.environ.get('KEY_ID')
+    role_arn = event['role_arn']
+    team_name = event['team_name']
+    parameter_prefix = "/{}/concourse/pipelines/{}".format(deployment,team_name)
+    try:
+        sts_response = sts.assume_role(
```

Comment:
I believe ([based on AWS' docs](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html)) that `assume_role` has a default `DurationSeconds` is 3600, would it be worth stating that explicitly here, or even reducing it - 900 seconds (15 mins) is the lowest?

Reply:
So the creds need to last long enough to be useful during a particular concourse job.  The creds get injected at the start of the job and won't get refreshed while the job runs, so we don't want them to expire during a longish terraform run.  I think instance profile creds (the current method) will default to 1 hour too so I don't think we're making things any worse.

Reply:
I'm going to open another PR to restrict role creds to only be usable from the concourse egress IPs, so that even if we leak the creds they shouldn't be usable.

Reply:
https://github.com/alphagov/tech-ops/pull/167